django/db/models/query.py

```diff
@@ -815,6 +815,25 @@ def complex_filter(self, filter_obj):
         else:
             return self._filter_or_exclude(None, **filter_obj)
 
+    def _combinator_query(self, combinator, *other_qs):
+        # Clone the query so we inherit the select list and everything
+        clone = self._clone()
+        # Clear limits and ordering, so they can be reapplied
+        clone.query.clear_ordering(True)
+        clone.query.clear_limits()
+        clone.query.combined_queries = (self.query,) + tuple(qs.query for qs in other_qs)
+        clone.query.combinator = combinator
+        return CombinedQuerySet(clone)
+
+    def union(self, *other_qs):
```

Comment:
@charettes How would a query like this look like in the ORM? I'd think that you'd do `TableExpression(your_union_query)` and then reuse that somewhere in the outer query -- so I do not see much of a potential for a conflict here. That said you could make the same argument for `order_by` and slicing, which you cannot move easily as arguments to `union` -- any ideas for those?